File: app/src/main/java/com/my/televip/obfuscate/ArgsResolver.java
package com.my.televip.obfuscate;


import com.my.televip.obfuscate.struct.ResolverRegistry;

import de.robv.android.xposed.XC_MethodHook;

public class ArgsResolver {

    public static ResolverRegistry resolverRegistry = new ResolverRegistry(ResolverRegistry.getResolverClass());

    public static Class<?>[] resolveObject(String name, Class<?>[] classes) {
        if (resolverRegistry != null) {
            if (resolverRegistry.hasParameter(name)) {
                return resolverRegistry.resolveParameter(name);
            }
        }

        return classes;
    }

    public static Object[] merge(String name, Class<?>[] classes, XC_MethodHook hook)
    {
        Class<?>[] c = resolveObject(name, classes);
        if (c != null) {
            Object[] result = new Object[c.length + 1];
            System.arraycopy(c, 0, result, 0, c.length);
            result[c.length] = hook;
            return result;
        }
        return null;
    }
}

File: app/src/main/java/com/my/televip/obfuscate/ArgsResolver.kt
package com.my.televip.obfuscate

import com.my.televip.obfuscate.struct.ResolverRegistry
import de.robv.android.xposed.XC_MethodHook

object ArgsResolver {

    @JvmField
    var resolverRegistry: ResolverRegistry = ResolverRegistry(ResolverRegistry.getResolverClass())

    @JvmStatic
    fun resolveObject(name: String, classes: Array<Class<*>>): Array<Class<*>> {
        if (resolverRegistry.hasParameter(name)) {
            val resolved = resolverRegistry.resolveParameter(name)
            if (resolved != null) {
                return resolved
            }
        }
        return classes
    }

    @JvmStatic
    fun merge(name: String, classes: Array<Class<*>>, hook: XC_MethodHook): Array<Any?>? {
        val c = resolveObject(name, classes)
        val result = arrayOfNulls<Any>(c.size + 1)
        System.arraycopy(c, 0, result, 0, c.size)
        result[c.size] = hook
        return result
    }
}
